.github/workflows/linux_sanitizers.yml

```diff
@@ -483,6 +494,90 @@ jobs:
         if: ${{ !cancelled() }}
         with:
-          name: test-results-cpp_${{ matrix.SANITIZER }}
+          name: unit-test-results-cpp_${{ matrix.SANITIZER }}
+          path: ${{ env.INSTALL_TEST_DIR }}/TEST*.xml
+          if-no-files-found: 'error'
+
+  CXX_Func_Tests:
+    # todo: move another plugin's func tests here
+    name: C++ func tests
+    if: ${{ github.repository_owner == 'openvinotoolkit' }}
+    needs: [Docker, Build]
+    timeout-minutes: 120
+    runs-on: aks-linux-16-cores-32gb
+    container:
+      image: ${{ fromJSON(needs.docker.outputs.images).ov_build.ubuntu_22_04_x64 }}
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      max-parallel: 3
+      fail-fast: false
+      matrix:
+        include:
+          - SANITIZER: 'AddressAndLeak'
+          - SANITIZER: 'UndefinedBehavior'
+          - SANITIZER: 'Thread'
+    env:
+      INSTALL_DIR: /__w/openvino/openvino/install
+      INSTALL_TEST_DIR: /__w/openvino/openvino/install/tests
+      
+    steps:
+      - name: Download OpenVINO package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ${{ format('openvino_package_{0}', matrix.SANITIZER) }}
+          path: ${{ env.INSTALL_DIR }}
+
+      - name: Download OpenVINO tests package
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16 # v4.1.8
+        with:
+          name: ${{ format('openvino_tests_{0}', matrix.SANITIZER) }}
+          path: ${{ env.INSTALL_TEST_DIR }}
+
+      - name: Extract OpenVINO packages
+        run: |
+          pushd $INSTALL_DIR
+            pigz -dc openvino_package.tar.gz | tar -xf - -C ${INSTALL_DIR}
+          popd
+          pushd $INSTALL_TEST_DIR
+            pigz -dc openvino_tests.tar.gz | tar -xf - -C ${INSTALL_DIR}
+          popd
+
+      - name: Fetch Sanitizer Suppression Lists
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        timeout-minutes: 15
+        with:
+          sparse-checkout: |
+            tests/sanitizers/lsan/suppressions.txt
+            tests/sanitizers/asan/suppressions.supp
+            tests/sanitizers/tsan/suppressions.txt
+          sparse-checkout-cone-mode: false
+          path: ${{ env.OPENVINO_REPO }}
+
+      #
+      # Tests
+      #
+
+      - name: CPU plugin func tests
+        id: cpu_func
+        # by default in scheduled launch we run only AddressAndLeak with *smoke*Conv_* pattern
+        # to prevent false-positive alerts and time exceeded error,
+        # in workflow_dispatch case all smoke tests will be launched with all types of sanitizers
+        # if gtestFilter and sanMode inputs are not specified
+        if: ${{ !cancelled() && (github.event_name == 'workflow_dispatch' && (inputs.sanitizerMode == 'All' || inputs.sanitizerMode == matrix.SANITIZER) || matrix.SANITIZER == 'AddressAndLeak')}}
```

Comment:
Please skip the entire workflow, as there is nothing else to test

Reply:
And we need to apply this condition for unit tests also.

Reply:
matrix context is out of visibility on the same level where matrix is defined, so I cannot check if ASAN type is going to be skipped on workflow / job level. The possible solution is to introduce preliminary step for the job, but in this case we have to introduce `needs` condition for every step, which is also obviously bad idea. I don't see another tradeoff in given case, than which is applied now.
This approach also means if-condition applied to every single step.

Reply:
But we can introduce something like dispatcher job, which implements matrix strategy and defines if the next job (connected by `needs`) is to be launch.
Job1: matrix...
Job2: 
needs Job1
if matrix.type == sometype